.github/workflows/check-python.yaml

```diff
@@ -8,16 +8,21 @@ jobs:
     runs-on: "ubuntu-latest"
     steps:
       - name: Checkout source code
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Install poetry
-        run: pipx install poetry
-
-      - name: Set up Python 3.10
+      - name: Set up Python 3.12
```

Comment:
This uses 3.12, but cd uses 3.10 - is that right?

Reply:
Yes, keeping cd as 3.10 to be on a safer side given that the cd wheel gets published to pypi

Reply:
Will revisit once follow up changes are merged